### ISSUE TITLE
Improvement to TrustID ecosystem

### ISSUE THREAD

**Issue by Zzocker** [hyperledger-labs/TrustID#40]
I have some query and suggestions regarding this project, specifically about the use of trustID in HL fabric application. I know one of the aims of this project is to provide a single digital identity for all types of blockchain.

Question:
Since invoke (worldstate update request) chaincode is not possible with chaincode installed on different channels. And trustID fabric chaincode uses `InvokeChaincode(chaincodeName string, args [][]byte, channel string) pb.Response` to invoke [services](https://github.com/hyperledger-labs/TrustID/blob/c1bec2c879a942d45927bace085c61903519c89a/fabric-chaincode/chaincode.gateway.go#L64). So to use dids stored on blockchain on different channels , one will have to install trustID chaincode to all these channels and manage the same dids documents on all these channels. To manage the same didis on many channels, trustID sdk will have to make atomic [requests](https://github.com/hyperledger-labs/TrustID/blob/c1bec2c879a942d45927bace085c61903519c89a/trustid-sdk/src/core/trustid.ts#L15) (CRUD on DIDs) to fabric. This atomic request won’t be easy to achieve. Please let me know if I am right till now, or am I missing something?

@mtnieto

**Comment by c20liu** [comment]
@Zzocker  
I agree with you that it is impossible to call updates across channels. If the application contract is updated, it can only be restricted to the same channel. In addition, the code seems to require that each application contract function be bound to a service, otherwise invoke will report an error

**Comment by c20liu** [comment]
@Zzocker   I am also learning the implementation of DID on Fabric, do you know what other projects can be studied, thank you
